transports/janus_websockets.c

```diff
@@ -790,17 +794,46 @@ static int janus_websockets_callback(struct libwebsocket_context *this,
 			}
 			if(!ws_client->destroy && !g_atomic_int_get(&stopping)) {
 				janus_mutex_lock(&ws_client->mutex);
-				/* Shoot all the pending messages first */
+				/* Check if we have a pending/partial write to complete first */
+				if(ws_client->partial_message && ws_client->partial_offset > 0
+						&& !ws_client->destroy && !g_atomic_int_get(&stopping)) {
+					int pending = strlen(ws_client->partial_message) - ws_client->partial_offset;
+					JANUS_LOG(LOG_VERB, "[WSS-%p] Completing pending WebSocket write (still need to write last %d/%zu bytes)...\n",
+						wsi, pending, strlen(ws_client->partial_message));
+					int sent = libwebsocket_write(wsi, (unsigned char *)(ws_client->partial_message + ws_client->partial_offset), pending, LWS_WRITE_TEXT);
```

Comment:
> The partial_message doesn't include the post-padding

That's done on purpose, as I followed this example [here](http://ml.libwebsockets.org/pipermail/libwebsockets/2014-November/001450.html): I guess the idea is that you're explicitly saying, this way, that this is part of a previous message you didn't complete, while padding would make it send a new frame, but not sure about this. Anyway it seemed to work as expected in my tests.

> I also wonder if it wouldn't be reasonable to pick a maximum size for buf, and make it an array on the stack, to avoid the extra malloc/free.

Since libwebsockets is single threaded and so you'd never write twice at the same time, it might be a good idea. I'm just hesitant at setting a maximum size, as in normal cases the only reason you might not send everything are TCP buffers, and so we might want to at least try pass everything we can. On the other hand, a static buffer to re-use would always have the safe padding, and would avoid the risk of writing over parts not allocated in a buffer. I'll give this a thought and a go later.